src/Directory.Build.props

```diff
@@ -69,9 +69,18 @@
 		<NetUWPOrWinUI Condition="'$(UNO_UWP_BUILD)'!='true'">net7.0-windows10.0.19041.0</NetUWPOrWinUI>
 	</PropertyGroup>
 
-	<PropertyGroup>
-		<MicrosoftNetCompilersToolsetVersion>4.6.0</MicrosoftNetCompilersToolsetVersion>
-	</PropertyGroup>
+	<Choose>
+		<When Condition="$(TargetFramework.StartsWith('net7.0'))">
+			<PropertyGroup>
+				<MicrosoftNetCompilersToolsetVersion>4.6.0</MicrosoftNetCompilersToolsetVersion>
+			</PropertyGroup>
+		</When>
+		<When Condition="$(TargetFramework.StartsWith('net8.0'))">
+			<PropertyGroup>
+				<MicrosoftNetCompilersToolsetVersion>4.7.0-2.final</MicrosoftNetCompilersToolsetVersion>
```

Comment:
Let's use the same compiler version in both cases. It is safe to upgrade to 4.7.0-2.final without conditions.